harness/determined/deploy/aws/cli.py

```diff
@@ -175,6 +179,16 @@ def deploy_aws(command: str, args: argparse.Namespace) -> None:
     if args.deployment_type != constants.deployment_types.LORE:
         if args.lore_version is not None:
             raise ValueError("--lore-version can only be specified for 'lore' deployments")
+    else:
+        print(
+            colored(
+                "Lore deployment type is experimental and not ready for production use.",
+                "yellow",
+            )
+        )
+        assert isinstance(args.lore_version, str)
+        if is_full_git_commit_hash(args.lore_version):
+            args.lore_version = args.lore_version[:7]
```

Comment:
Why would we want to fail here?

Reply:
I take that you mean just failing if they pass the entire hash as opposed to truncating and then pull docker based on truncated hash.

Concern:
- That would be different from the regular det deploy behavior. 

If we want to fail, I think a better place is if the docker image cannot be found, then we should fail with the right message. Right now, it would not tell you Lore deployment failed and you need to go to the instance to see there is no docker for lore running.

Reply:
Maybe print a warning telling them using full hash is not correct and we are truncating?